src/EditModel/EditDataEntryForm/Palette.component.js

```diff
@@ -0,0 +1,112 @@
+import React, { useState, useRef } from 'react';
+import PropTypes from 'prop-types';
+import TextField from 'material-ui/TextField/TextField';
+import CheckBox from 'material-ui/Checkbox/Checkbox';
+import PaletteSection from './PaletteSection.component';
+import Action from 'd2-ui/lib/action/Action';
+
+function useConst(factory) {
+  const ref = useRef(null)
+  if (ref.current === null) {
+    ref.current = factory()
+  }
+  return ref.current
+}
+
+const styles = {
+    paletteFilter: {
+        position: 'absolute',
+        top: -16,
+        width: '100%',
+        padding: '8px 8px 16px',
+    },
+    paletteFilterField: {
+        width: '100%',
+    },
+    greySwitch: {
+        position: 'absolute',
+        bottom: 8,
+        left: 8,
+        right: 8,
+    },
+}
+
+const Palette = ({
```

Comment:
This was extracted from https://github.com/dhis2/maintenance-app/blob/224f2ec40242776d1e92ab5d6fdc2cc9d4989c88/src/EditModel/EditDataEntryForm.component.js#L371-L403